core/tests/protractor_utils/ExplorationPlayerPage.js

```diff
@@ -170,17 +170,29 @@ var ExplorationPlayerPage = function() {
   };
 
   this.submitFeedback = function(feedback) {
-    feedbackPopupLink.click();
+    feedbackPopupLink.isDisplayed().then(function(){
+      feedbackPopupLink.click();
+    });
     feedbackTextArea.sendKeys(feedback);
-    feedbackSubmitButton.click();
+    feedbackSubmitButton.isDisplayed().then(function() {
+      feedbackSubmitButton.click();
+    });
+    general.waitForSystem();
   };
 
   this.submitSuggestion = function(suggestion, description) {
-    suggestionPopupLink.click();
+    suggestionPopupLink.isDisplayed().then(function(){
+      suggestionPopupLink.click();
+    });
+    browser.waitForAngular();
```

Comment:
Are all these waitForAngular() calls necessary? Better to avoid if possible, but I understand putting them in if the specific part of the test is flaky. But we should only do it for those parts.

Reply:
This looks like an absolute overdose of `waitForAngular`. I have removed them and put a single `waitForSystem` at the very end 👍

Reply:
Just a quick question : do you think page object workflow should contain `waitForSystem` or `waitForAngular` calls? 
I think test are flaky because of sequences of actions performed too quickly (user logs out immediately after clicking submit suggestion) so `waitForSystem` or `waitForAngular` should only be reserved for use in the 
`it("should...") {`
specs. What do you think?